cache_busting/mod.ts

```diff
@@ -0,0 +1,103 @@
+import binaryLoader from "lume/core/loaders/binary.ts";
+import { getPathAndExtension, merge } from "lume/core/utils.ts";
+import { encode } from "lume/deps/hex.ts";
+import { posix } from "lume/deps/path.ts";
+import modifyUrls from "lume/plugins/modify_urls.ts";
+
+import type { Page, Plugin, Site } from "lume/core.ts";
+import type { Element } from "lume/deps/dom.ts";
+
+export interface Options {
+  /** Attribute used to select the elements this plugin applies to */
+  attribute: string;
+
+  /** The length of file hashes to generate */
+  hashLength: number;
+}
+
+// Default options
+export const defaults: Options = {
+  attribute: "hash",
+  hashLength: 10,
+};
+
+const cache = new Map<string, string>();
+
+/** A plugin to add cache busting hashes to all URLs found in HTML documents. */
+export default function (userOptions?: Partial<Options>): Plugin {
+  const options = merge(defaults, userOptions);
+
+  return (site: Site) => {
+    const selector = `[${options.attribute}]`
+
+    site.use(modifyUrls({ fn: replace }));
+
+    async function replace(url: string | null, _: Page, element: Element) {
+      if (url && element.matches(selector)) {
+        return await addHash(url);
+      }
+  
+      return "";
+    }
+
+    async function addHash(url: string) {
+      // Ensure the path starts with "/"
+      url = posix.join("/", url);
```

Comment:
This won't work for relative urls like `./image.png`.
Probably you should do something like (didn't test it):

```js
if (url.startsWith(".")) { // Is a relative url
  url = posix.join(page.data.url, url);
}
```